src/Compiler/Microsoft.AspNetCore.Razor.Language/src/RazorCodeDocumentExtensions.cs

```diff
@@ -292,7 +292,7 @@ public static bool TryComputeClassName(this RazorCodeDocument codeDocument, [Not
             return false;
         }
 
-        className = CSharpIdentifier.GetClassNameFromPath(filePath);
+        className = CSharpIdentifier.GetClassNameFromPath(filePath!);
```

Comment:
As an alternative, consider using the `IsNullOrEmpty` extension method from `Utilities.Shared` above, instead of `string.IsNullOrEmpty`. That method is nullable annotated, so the `!` would be redundant here.